src/services/utilServices/SendDNSRecordEmailClient.ts

```diff
@@ -1,5 +1,11 @@
-import { groupBy } from "lodash"
+import { groupBy, has } from "lodash"
 
+export interface QuadARecord {
+  domain: string
+  class: string
+  type: string
```

Comment:
I wanted to decouple the two "interfaces". the email client should not be coupled with the digtype etc.. rather its job is to just receive a string and inject it.

but for the dig function it should have a more contextual type on the values it can and should be receiving.

wdyt?

Reply:
i think this is ok but would `QuadARecord.type` ever be different from `DigType`?

Reply:
It would never be different unless we want to send in a string with formatting like .toLowerCase/UpperCase etc.

That's why kept this as string in case we want to inject a formatted DigType instead of the actual enum

Reply:
hmm, actually this is a valid use case and typescript exports [certain string literal types](https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html#uppercasestringtype) for exactly this! setting this as a type would also allow us to enforce said casing also hehe

Reply:
Okies, makes sense! Extracted the relevant types to separate file under `types/` as well